sbi/inference/posteriors/sbi_posterior.py

```diff
@@ -157,7 +183,11 @@ def _log_prob_snpe(self, theta: Tensor, x: Tensor, norm_posterior: bool) -> Tens
         # Force probability to be zero outside prior support.
         masked_log_prob = torch.where(is_prior_finite, unnorm_log_prob, NEG_INF)
 
-        log_factor = log(self.get_leakage_correction(x=x)) if norm_posterior else 0
+        log_factor = (
```

Comment:
I generally agree, but in this specific case I kind of like it. Just because the first two lines read so nicely, and they are what will be used mostly.